vnext/Microsoft.ReactNative/IReactDispatcher.h

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+#include "ReactDispatcherHelper.g.h"
+#include <ReactPropertyBag.h>
+#include <dispatchQueue/dispatchQueue.h>
+#include <winrt/Microsoft.ReactNative.h>
+
+namespace winrt::Microsoft::ReactNative {
+
+struct ReactDispatcher : implements<ReactDispatcher, IReactDispatcher> {
+  ReactDispatcher() = default;
+  ReactDispatcher(Mso::DispatchQueue &&queue) noexcept;
+
+  bool HasThreadAccess() noexcept;
+  void Post(ReactDispatcherCallback const &callback) noexcept;
+
+  static Mso::DispatchQueue GetUIDispatchQueue(IReactPropertyBag const &properties) noexcept;
+
+  static IReactDispatcher UIThreadDispatcher() noexcept;
+  static ReactPropertyId<IReactDispatcher> UIThreadDispatcherProperty() noexcept;
+  static IReactDispatcher GetUIThreadDispatcher(IReactPropertyBag const &properties) noexcept;
+  static void SetUIThreadDispatcher(IReactPropertyBag const &properties) noexcept;
```

Comment:
>SetUIThreadDispatcher [](start = 14, length = 21)

this is weird, why not just have SetUIThreadDispatcher(IReactDispatcher)?
that way users can do
SetUIThreadDispatcher(GetUIThreadDispatcher(properties))
or am I misunderstanding this Api?